lib/rucio/core/replica.py

```diff
@@ -4472,3 +4496,84 @@ def get_RSEcoverage_of_dataset(
             result[rse_id] = total
 
     return result
+
+
+@transactional_session
+def refresh_replicas(
+        rse_id: Optional[str] = None,
+        replicas: Optional['Iterable[dict[str, Any]]'] = None,
+        *,
+        session: "Session"
+) -> bool:
```

Comment:
I would have prepared to see the last part of `list_and_mark_unlocked_replicas()` factored out into a separate function. Is that not feasible?